api_branch_restrictions.go

```diff
@@ -566,127 +567,42 @@ Creates a new branch restriction rule for a repository.  &#x60;kind&#x60; descri
 
 @return Branchrestriction
 */
-func (a *BranchRestrictionsApiService) RepositoriesWorkspaceRepoSlugBranchRestrictionsPost(ctx context.Context, body Branchrestriction, repoSlug string, workspace string) (Branchrestriction, *http.Response, error) {
-	var (
-		localVarHttpMethod  = strings.ToUpper("Post")
-		localVarPostBody    interface{}
-		localVarFileName    string
-		localVarFileBytes   []byte
-		localVarReturnValue Branchrestriction
-	)
-
-	// create path and map variables
-	localVarPath := a.client.cfg.BasePath + "/repositories/{workspace}/{repo_slug}/branch-restrictions"
-	localVarPath = strings.Replace(localVarPath, "{"+"repo_slug"+"}", fmt.Sprintf("%v", repoSlug), -1)
-	localVarPath = strings.Replace(localVarPath, "{"+"workspace"+"}", fmt.Sprintf("%v", workspace), -1)
-
+func (a *BranchRestrictionsApiService) RepositoriesWorkspaceRepoSlugBranchRestrictionsPost(ctx context.Context, body Branchrestriction, repoSlug string, workspace string) (*Branchrestriction, *http.Response, *GenericSwaggerError, error) {
+	localVarPath := fmt.Sprintf("%v/repositories/%v/%v/branch-restrictions", a.client.cfg.BasePath, workspace, repoSlug)
 	localVarHeaderParams := make(map[string]string)
-	localVarQueryParams := url.Values{}
-	localVarFormParams := url.Values{}
+	localVarHeaderParams["Accept"] = "application/json"
 
-	// to determine the Content-Type header
-	localVarHttpContentTypes := []string{"application/json"}
-
-	// set Content-Type header
-	localVarHttpContentType := selectHeaderContentType(localVarHttpContentTypes)
-	if localVarHttpContentType != "" {
-		localVarHeaderParams["Content-Type"] = localVarHttpContentType
-	}
-
-	// to determine the Accept header
-	localVarHttpHeaderAccepts := []string{"application/json"}
-
-	// set Accept header
-	localVarHttpHeaderAccept := selectHeaderAccept(localVarHttpHeaderAccepts)
```

Comment:
I do not completely understand this complexity, and changed this to set this directly in the `HeaderParams`. Let me know if I missed something.